src/index.js

```diff
@@ -0,0 +1,158 @@
+    tempDecreaseButton = document.getElementById('decreaseTempControl');
+    tempValue = document.getElementById('tempValue');
+    gardenContent = document.getElementById('landscape');
+    headerCity = document.getElementById('headerCityName');
+    cityNameUserInput = document.getElementById('cityNameInput');
+    skyInput = document.getElementById('skySelect');
+    skyTemplate = document.getElementById('sky');
+    resetCityName = document.getElementById('cityNameReset');
+    realTimeTemperature = document.getElementById('currentTempButton');
+};
+//Create optionsa element for sky in the HTML file
+const createOptions = () => {
+    const optionArray = ['Sunny', 'Cloudy', 'Snowy', 'Rainy'];
+    for (let i = 0; i < optionArray.length; ++i) {
+        skyOption = document.createElement('option');
+        skyOption.textContent = optionArray[i];
+        skyInput.appendChild(skyOption);
+    }
+};
+
+const registerEvent = () => {
+    tempIncreaseButton.addEventListener('click', handleIncreaseTempClick);
+    tempDecreaseButton.addEventListener('click', handleDecreaseTempClick);
+    resetCityName.addEventListener('click', handleResetButton);
+    realTimeTemperature.addEventListener('click', handleRealTemp);
+    skyInput.addEventListener('change', skyInputHandler);
+    cityNameUserInput.addEventListener('input', headerCityFunction);
+};
+
+const handleIncreaseTempClick = () => {
+    ++temperature;
+    showTemp();
+};
+
+const handleDecreaseTempClick = () => {
+    --temperature;
+    showTemp();
+};
+
+
+
+const showTemp = () => {
+    tempValue.textContent = temperature;
+    updateTempColor();
+};
+
+const headerCityFunction = () => {
+    headerCity.textContent = cityNameUserInput.value; //defaultCityName;
+};
+
+const skyInputHandler = () => {
+    const input = skyInput.value;
+    skyTemplate.textContent = sky[input];
+};
+
+const handleResetButton = () => {
+    cityNameUserInput.value = '';
+    headerCity.textContent = '';
+};
+
+// wave 4 - calling API
+const handleRealTemp = async () => {
+    tempValue.textContent = Math.floor((await getWeather() - 273.15) * 9/5 + 32 );
+    temperature = tempValue.textContent
+    updateTempColor();
+};
+
+const getLanLon = async (placeName) => {
+    const response = await axios.get('http://127.0.0.1:5000/location', {
+        params: {
+            q: placeName,
+        },
+    });
+    const { lat: latitude, lon: longitude } = response.data[0];
+    return { latitude, longitude };
+};
+const getWeather = async () => {
+    const { latitude, longitude } = await getLanLon(cityNameUserInput.value);
+    const response = await axios.get('http://127.0.0.1:5000/weather', {
+        params: {
+            lat: latitude,
+            lon: longitude,
+        },
+    });
+    return response.data.main.temp;
+};
+
+const updateTempColor = () => {
+    if (temperature >= 80) {
+        tempValue.classList.add('red');
+        tempValue.classList.remove('orange', 'yellow', 'yellow-green', 'teal');
```

Comment:
Great use of add and remove to ensure we aren't clearing any other classes that might be set on the element!